crates/volta-core/src/hook/serial.rs

```diff
@@ -75,6 +84,26 @@ impl RawResolveHook {
     }
 }
 
+impl RawIndexHook {
+    pub fn into_index_hook(self, base_dir: &Path) -> Fallible<tool::YarnIndexHook> {
+        // use user-specified format, or default to Github (legacy)
+        let format = if let Some(format_str) = self.format {
+            RegistryFormat::from_str(&format_str)?
+        } else {
+            RegistryFormat::Github
+        };
```

Comment:
Tiny nit: I typically find it a little more readable to use `match` instead of `if let { } else { }`, something like:

```rust
let format = match self.format {
    Some(format_str) => RegistryFormat::from_str(&format_str)?,
    None => RegistryFormat::Github,
};
```

Another option would be to use the more "fluent" methods on `Option`, but given the need to early return with an error, that may wind up looking _less_ readable / maintainable:

```rust
let format = self.format
    .map(|format_str| RegistryFormat::from_str(&format_str))
    .transpose()?
    .unwrap_or(RegistryFormat::Github);
```